src/main/java/org/icgc/argo/program_service/services/AuthorizationService.java

```diff
@@ -0,0 +1,55 @@
+package org.icgc.argo.program_service.services;
+
+import io.grpc.Status;
+
+public interface AuthorizationService {
+  boolean isDCCAdmin();
+  boolean hasPermission(String permission);
+  boolean hasEmail(String email);
+
+  private String readPermission(String programShortName) {
+    return "PROGRAM-"+ programShortName +".READ";
+  }
+
+  private String writePermission(String programShortName) {
+    return "PROGRAM-" + programShortName + ".WRITE";
+  }
+
+  default void require(boolean condition) {
```

Comment:
I like the idea of a checker method that throws on false, but you need to be able to provide a message for the logs so you can log right before you throw. 

`require(boolean condition, String message)`